text/WildcardMatching/Wildcard Matching - Kamil Rajtar.tex

```diff
@@ -0,0 +1,64 @@
+\documentclass{article}
+\usepackage[utf8]{inputenc}
+
+\title{Wyszukiwanie z wildcardami w tekście i wzorcu}
+\author{Kamil Rajtar}
+\date{Czerwiec 2020}
+
+\begin{document}
+
+\newcommand\defeq{\stackrel{\mathclap{\normalfont\mbox{def}}}{=}}
+
+\maketitle
+
+\section*{Opis problemu}
+
+Na wejściu do algorytmu dostajemy tekst i wzorzec w których oprócz liter mogą występować wildcardy (?) pasujące do dowolnego znaku. Na wyjściu ma się znaleźć lista pozycji w których wzorzec pasuje do tekstu.
+
+\section*{Wersja bez wildcardów}
+
+W rozwiązaniu problemu użyty jest splot obliczany za pomocą szybkiej transformaty Furiera (FFT) zdefiniowany następująco:
+
+$$
+    p \otimes t \defeq (\sum^{m-1}_{j=0}p_{j}t_{i+j}, 0 \leq i \leq n-m)
+$$
+
+Normalne (bez wildcardów) dopasowanie za pomocą splotu tekstu (t) i odwróconego wzorca (p) możemy obliczyć za pomocą wzoru:
```

Comment:
Probably it's worth mentioning that you mean matching between $t[i:i + m]$ and $p$.